src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -440,39 +440,52 @@ private static void InitializeGetEnumerableCache()
 
         internal static bool IsTypeEnumerable(Type type)
         {
+            if (type == null) { return false; }
             GetEnumerableDelegate getEnumerable = GetOrCalculateEnumerable(type);
             return (getEnumerable != LanguagePrimitives.ReturnNullEnumerable);
         }
 
-        /// <summary>
-        /// Retrieves the IEnumerable of obj or null if the language does not consider obj to be IEnumerable
-        /// </summary>
-        /// <param name="obj">
-        /// IEnumerable or IEnumerable-like object
-        /// </param>
-        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
-        public static IEnumerable GetEnumerable(object obj)
+        private static object GetBaseObject(object obj)
         {
             if (obj == null)
             {
                 return null;
             }
 
-            Type objectType = obj.GetType();
-
-            // if the object passed is an PSObject,
-            // look at the base object. Notice that, if the
-            // object has been serialized, the base object
-            // will be there as an ArrayList if the original
-            // object was IEnumerable
-            if (objectType == typeof(PSObject))
+            var psobj = obj as PSObject;
+            if (null != psobj)
             {
-                PSObject mshObj = (PSObject)obj;
-                obj = mshObj.BaseObject;
-                objectType = obj.GetType();
+                obj = psobj.BaseObject;
             }
 
-            GetEnumerableDelegate getEnumerable = GetOrCalculateEnumerable(objectType);
+            return obj;
+        }
+
+        /// <summary>
+        /// Returns True if the language considers obj to be IEnumerable
+        /// </summary>
+        /// <param name="obj">
+        /// IEnumerable or IEnumerable-like object
+        /// </param>
+        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
+        public static bool IsObjectEnumerable(object obj)
+        {
+            return IsTypeEnumerable(GetBaseObject(obj)?.GetType());
+        }
+
+
+        /// <summary>
+        /// Retrieves the IEnumerable of obj or null if the language does not consider obj to be IEnumerable
+        /// </summary>
+        /// <param name="obj">
+        /// IEnumerable or IEnumerable-like object
+        /// </param>
+        [SuppressMessage("Microsoft.Naming", "CA1720:IdentifiersShouldNotContainTypeNames", MessageId = "obj", Justification = "Since V1 code is already shipped, excluding this message.")]
+        public static IEnumerable GetEnumerable(object obj)
+        {
+            obj = GetBaseObject(obj);
```

Comment:
How about replace `GetBaseObject(object)` with [`PSObject.Base(object)`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/MshObject.cs#L989)? It seems a good fit here.

Reply:
I just knew there was already going to be a method for this somewhere in the code.  :)  I searched for .BaseObject and didn't find it; Base() uses the field name instead.